[Jun Rong] iP

kerburenthusiasm opened the pull request.

Commit:
Add Gradle support

Commit:
Update Duke.java, README.md, runtest.bat

Update the aforementioned files to better reflect on the current
project. Any mentioned of "Duke" was changed to "FunBox".

Commit:
Add FunBoxGear.java

FunBoxGear.java contains the functionalities which will be used by
FunBox.

It contains basic commands such as,

* Greet
* Echo
* Exit

Commit:
Update FunBoxGear.java and FunBox.java

Update FunBoxGear.java with functionalities to store user's messages
and display these messages in a list.

Subsequently, FunBox.java is updated to use the new functionalities
implemented.

Commit:
Add Task.java

Add Task.java for better encapsulation under OOP principles.

Commit:
Update Task.java

Update Task.java to reflect that the code was reused with
modifications.

Commit:
Update FunBoxGear.java

Update FunBoxGear.java to follow closely to the OOP principle.

Functions such as isBye, isList etc. used to detect the type of
message sent by the users has similar behaviour resulting in code
duplication.

The getCommands() function remove such duplication by combining the
aforementioned functions together.

FunBoxGear.java has also been updated to make use of Task.java.

Commit:
Update FunBox.java

Update FunBox.java to utilize the updated FunBoxGear.java. Multiple
if-else statements has been reduced to a single if statements.

Commit:
Add Deadline.java, Event.java, ToDo.java

Deadline, Event, ToDo are different type of tasks with some
similarity between them.

Due to the similarity, they all inherit from Task.java. Inheritance
is used to follow the OOP principle: polymorphism.

Commit:
Update FunBoxGear.java

Update FunBoxGear.java to make use of the newly added files:

* Deadline.java
* Event.java
* Task.java

This allows users to categorize the different type of tasks.

Commit:
Update FunBoxGear.java

Removed emoji due unmappable character error when running test.

Commit:
Update EXPECTED.TXT and input.txt

Update EXPECTED.TXT and input.txt such that the current input and
result can be used as a benchmark to compare to when the code
changes in the future.

Commit:
Add FunBoxExceptions.java

Add FunBoxExceptions.java to add error handlings for FunBox.

Adding custom exceptions allows for handling of specific exceptions
related to FunBox

Commit:
Update FunBoxGear.java

Update FunBoxGear.java to utilize FunBoxExceptions' functionalities
for error handling.

Commit:
Update FunBoxGear.java

Update FunBoxGear.java to allow users to delete tasks they inputted.

Commit:
Update FunBoxGear.java

Update FunBoxGear.java to include exceptions for mark, unmark, and
delete functions.

This is to provide a more comprehensive error handling such that
users would understand where their command entered went wrong.

Commit:
Update EXPECTED.TXT

Update EXPECTED.TXT on Line 1 as it expects a blank space at the
end of the sentence.

Commit:
Update FunBoxGear.java, Task.java

Commit:
Updated Task.java and its children classes

Commit:
Revert "Updated Task.java and its children classes"

This reverts commit ba391cd98c5802634c8f0e70cd3425127b21d97d.

Commit:
Update Task.java and its children

Commit:
Update FunBoxGear.java

Commit:
Merge branch 'branch-Level-7'

Commit:
Update FunBoxGear.java and Task.java to resolve merge conflict

Commit:
Merge branch 'branch-Level-8'

Commit:
Refactor code

Commit:
Add `filter` command

Commit:
Split classes into packages

Commit:
Add jUnit tests

Commit:
Add find command

Commit:
Adds JavaDoc for classes

Commit:
Merge branch 'branch-Level-9'

Commit:
Resolve merge conflicts for A-CodingStandard, A-JavaDoc, Level-9

Commit:
Merge branch 'add-gradle-support' into branch-A-Gradle

Commit:
Add Gradle and checkstyle

Commit:
Merge branch 'add-gradle-support' into branch-GUI

Commit:
Added GUI

Commit:
Merge branch 'branch-A-Gradle'

Commit:
Merge branch 'branch-GUI'

Commit:
Remove data folder

RyanCheungJF commented on src/main/java/funbox/util/TaskList.java:
```diff
@@ -0,0 +1,193 @@
+import funbox.task.Event;
+
+public class TaskList {
+    private ArrayList<Task> taskList;
```
Not too sure about this but I think ArrayList is considered under Collections, and the coding standards ask for a plural form on names representing a collection. The naming sounds ok since `taskLists` would sound weird, but maybe consider `tasks`? Although if it causes ambiguity it might be good to check if `taskList` is accepted!

RyanCheungJF commented on src/main/java/funbox/FunBox.java:
```diff
@@ -0,0 +1,64 @@
+     */
+    public Result getResponse(String input) {
+        String result = "";
+        boolean isExit = false;
```
Good use of boolean names, adheres to the coding standard!

RyanCheungJF commented on src/main/java/funbox/Main.java:
```diff
@@ -0,0 +1,38 @@
+    private FunBox funBox = new FunBox();
+
+    /**
+     * Display the GUI for the user.
```
I believe u have to use the verb form, as per the [coding standard](https://se-education.org/guides/conventions/java/intermediate.html#comments):

> In method header comments, the first sentence should start in the form Returns ..., Sends ..., Adds ... (not Return or Returnning etc.)

```suggestion
     * Displays the GUI for the user.
```

RyanCheungJF commented on src/main/java/funbox/task/Task.java:
```diff
@@ -0,0 +1,66 @@
+     * @return Return "X" if status of task is completed else return " "
+     */
+    public String getStatusIcon() {
+        return (isDone ? "X" : " "); // mark done task with X
```
Not sure how strict they are, but the coding standards provide these 2 formats as acceptable:

> Here are two acceptable ways to format ternary expressions:

```java
alpha = (aLongBooleanExpression) ? beta : gamma;
alpha = (aLongBooleanExpression)
        ? beta
        : gamma;
```

```suggestion
        return (isDone) ? "X" : " "; // mark done task with X
```

RyanCheungJF reviewed (commented):
Overall looks good to me! Did a great job and may only need minor changes with the most significant one being Javadocs. Best of luck!

kevinkuo0320 commented on src/main/java/funbox/MainWindow.java:
```diff
@@ -0,0 +1,81 @@
+    @FXML
+    private Button sendButton;
+
+    private FunBox funbox;
```
Good use of modifier!

kevinkuo0320 commented on src/main/java/funbox/util/Parser.java:
```diff
@@ -0,0 +1,151 @@
+ */
+public class Parser {
+
+    public final String unknownCommand = "Sorry! You have used the command wrongly!";
```
coding style - I think it might be better to use capital letters for the final string?

kevinkuo0320 reviewed (commented):
Overall excellent work!

Commit:
Update files

Commit:
Add assertions to validate assumptions

There are several methods which make assumptions about the input
provided by assuming they are valid.

Let's add assertions to methods which made assumptions
about the state of the input at certain points.

Using assertions helps document important assumptions that should
hold at various points in the code.

Commit:
Remove unnecessary assertions

Commit:
Refactor files to improve code quality.

Duplication of code and dead code were found throughout
the file indicating poor implementation of codes.

Extracting duplication into a single function allows us to use
abstraction. Additionally, deleting dead code improves readability
of code.

Let's reduce duplication of codes by creating a function for it
and removes dead codes found.

kerburenthusiasm committed:
Merge pull request #1 from junrong98/branch-A-Assertions

Add assertions to validate assumptions

kerburenthusiasm committed:
Merge branch 'master' into branch-A-CodeQuality

kerburenthusiasm committed:
Merge pull request #2 from junrong98/branch-A-CodeQuality

Refactor files to improve code quality.

Commit:
Fix small issues

Commit:
Add tagging functionality to task.

Let's add the ability for users to tag tasks.

Commit:
Merge branch 'branch-C-Tagging'

Commit:
Fix minor issue related to tagging.

Commit:
Update UI to look more aesthetic.

Commit:
Update README.md.

Includes features of FunBox and Usage of FunBox.

Commit:
Remove Ui.PNG

Commit:
Add Ui.png

Commit:
Update README.md

kerburenthusiasm committed:
Set theme jekyll-theme-minimal

kerburenthusiasm committed:
Set theme jekyll-theme-minimal

Commit:
Fix minor issues.

Commit:
Merge branch 'master' of https://github.com/junrong98/ip

Commit:
Fix minor issues.
